core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -498,6 +505,49 @@ protected Optional<Expression> translateInPredicate(ConnectorExpression value, C
             return Optional.empty();
         }
 
+        protected Optional<Expression> translateSimpleCaseWhen(List<ConnectorExpression> arguments)
+        {
+            int argumentsSize = arguments.size();
+            if (argumentsSize < 3) {
+                return Optional.empty();
+            }
+
+            Optional<Expression> operand = translate(arguments.get(0));
+            if (operand.isEmpty()) {
+                return Optional.empty();
+            }
+
+            ImmutableList.Builder<WhenClause> builder = ImmutableList.builder();
+            for (int i = 1; i < argumentsSize - 1; i += 2) {
```

Comment:
This seems complex. Can we model it in the following way instead:

`$simple_case_when(operand, $array($simple_case_when_condition(condition, value)), [defaultValue])`

@findepi thoughts?

Reply:
> This seems complex. Can we model it in the following way instead:
> 
> `$simple_case_when(operand, $array($simple_case_when_condition(condition, value)), [defaultValue])`
> 
> @findepi thoughts?

This is a great idea, and I will do some exploration.